src/LibraryManager/CacheService/WebRequestHandler.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager
+{
+    /// <summary>
+    /// Helper class to hold the HttpClient instance and send requests to get resources
+    /// </summary>
+    internal class WebRequestHandler : IWebRequestHandler, IDisposable
+    {
+        private HttpClient _httpClient;
+        private WebRequestHandler()
+        {
+            HttpClientHandler httpMessageHandler = new HttpClientHandler();
+            _httpClient = new HttpClient(httpMessageHandler);
+        }
+
+        public static IWebRequestHandler Instance { get; } = new WebRequestHandler();
+
+        public void Dispose()
```

Comment:
Should this implement the disposable pattern with the boolean flag?

Reply:
Spoke to @raquelsa, we want to keep this class as a singleton. So going to remove the IDisposable for now.